File: app/src/main/java/com/btntrung/pointmanagement/ManagerMainActivity.java
package com.btntrung.pointmanagement;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.btntrung.pointmanagement.adapter.ManagerSearchClassAdapter;
import com.btntrung.pointmanagement.adapter.ManagerSearchStudentAdapter;
import com.btntrung.pointmanagement.entity.ClassRoom;
import com.btntrung.pointmanagement.entity.Student;

import java.util.ArrayList;
import java.util.List;

public class ManagerMainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private Button btn_search;
    private TextView textView;
    private EditText key;
    private ManagerSearchClassAdapter adapter;
    private List<ClassRoom> classRooms=new ArrayList<>();


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_manager_main);

        recyclerView=findViewById(R.id.recycle_view);
        btn_search=findViewById(R.id.btn_search);
        key=findViewById(R.id.edit_search);
//        recyclerView.setHasFixedSize(true);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));

        ClassRoom classRoom=new ClassRoom(1,"e17cn02");
        ClassRoom classRoom1=new ClassRoom(2,"e17cn03");

        classRooms.add(classRoom);
        classRooms.add(classRoom1);

        btn_search.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name=key.getText().toString();

                adapter=new ManagerSearchClassAdapter(ManagerMainActivity.this,classRooms);
                recyclerView.setAdapter(adapter);
            }
        });

    }
}

File: app/src/main/java/com/btntrung/pointmanagement/presentation/manager/ManagerMainActivity.kt
package com.btntrung.pointmanagement.presentation.manager

import android.content.Intent
import android.view.Menu
import android.view.MenuItem
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.LinearLayoutManager
import com.btntrung.pointmanagement.R
import com.btntrung.pointmanagement.adapter.ManagerSearchClassAdapter
import com.btntrung.pointmanagement.databinding.ActivityManagerMainBinding
import com.btntrung.pointmanagement.entity.Classroom
import com.btntrung.pointmanagement.entity.Semester
import com.btntrung.pointmanagement.presentation.manager.ClassroomListAdapter
import com.btntrung.pointmanagement.presentation.manager.ManagerMainViewModel
import com.btntrung.pointmanagement.presentation.student.StudentMainActivity
import com.btntrung.pointmanagement.ProfileActivity
import kotlinx.coroutines.flow.collect
import org.koin.androidx.viewmodel.ext.android.viewModel
import com.google.firebase.auth.FirebaseAuth
import timber.log.Timber
import java.util.*

class ManagerMainActivity : AppCompatActivity() {
    private val viewModel: ManagerMainViewModel by viewModel()

    private lateinit var binding: ActivityManagerMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_manager_main)

        val user = FirebaseAuth.getInstance().currentUser
        user?.let {
            for (profile in it.providerData) {
                val providerId = profile.providerId
                val uid = profile.uid
                val name = profile.displayName
                val email = profile.email
                val photoUrl = profile.photoUrl

                val actionBar = supportActionBar
                actionBar!!.title = name
            }
        }

    }
    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        val id = item.getItemId()

        if (id == R.id.logout) {
//            Toast.makeText(this, "Item Clicked", Toast.LENGTH_LONG).show()
            return true
        }
        if (id == R.id.profile) {
            val intent:Intent=Intent(this,ProfileActivity::class.java)
            startActivity(intent)
            return true
        }

        return super.onOptionsItemSelected(item)

    }

}